src/providers/textEditorBasedProvider.ts

```diff
@@ -0,0 +1,12 @@
+import { TextEditor, window } from "vscode";
+import { Provider } from "./provider";
+
+export class TextEditorBasedProvider extends Provider {
+    protected get textEditor() {
+        return window.activeTextEditor;
+    }
```

Comment:
TextEditor is unused.
```suggestion
import { type TextEditor, window } from "vscode";
import { Provider } from "./provider";

export class TextEditorBasedProvider extends Provider {
    protected get textEditor(): TextEditor | undefined {
        return window.activeTextEditor;
    }
```